src/main.rs

```diff
@@ -46,26 +46,14 @@ Use MK_LOG=trace to see debug output.
         );
         exit(0);
     }
-    match Newer::new(args) {
-        Ok(newer) => {
-            if !newer.should_run_command() {
-                debug!("Nothing to do.");
-                exit(if newer.newer { 1 } else { 0 });
-            } else {
-                match run_command(newer.command) {
-                    Ok(_) => {}
-                    Err(e) => {
-                        eprintln!("{}", e);
-                        exit(1);
-                    }
-                }
-            }
-        }
-        Err(e) => {
-            eprintln!("{}", e);
-            exit(1);
-        }
-    }
+    let newer = Newer::new(args)?;
+    let code = if newer.should_run_command() {
+        run_command(newer.command)?
```

Comment:
**Behavior change**

In the original code, the exit code from `run_command` was swallowed and replaced by `0`.

Reply:
Ah good catch, fixed that in main.